internal/event-processor/repo.go

```
@@ -78,6 +78,26 @@ func updateSystem(ctx context.Context, r repo.Repo, system *model.System) error
+	ck := repo.NewCompositeKey().Where(fmt.Sprintf("%s_%s", repo.KeyField, repo.IDField), keyID)
+	query := repo.NewQuery().
+		Where(repo.NewCompositeKeyGroup(ck)).
+		Order(repo.OrderField{Field: repo.CreatedField, Direction: repo.Desc})
+
+	_, err := r.First(ctx, &kv, *query)
+	if err != nil {
+		return "", fmt.Errorf("failed to get newest key version for key %s: %w", keyID, err)
+	}
+
+	if kv.NativeID == nil {
+		return "", fmt.Errorf("newest key version for key %s has no native ID", keyID)
```

golangci-lint (2.11.4)

[error] do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"newest key version for key %s has no native ID\", keyID)"

(err113)